src/ibex_bluesky_core/callbacks/file_logger.py

```diff
@@ -79,9 +79,11 @@ def start(self, doc: RunStart) -> None:
         )
         header_data[START_TIME] = formatted_time
 
+        if not self.filename.parent.exists():
+            self.filename.parent.mkdir()
+
         with open(self.filename, "a", newline="", encoding="utf-8") as outfile:
-            for key, value in header_data.items():
-                outfile.write(f"{key}: {value}\n")
+            outfile.writelines([f"{key}: {value}\n" for key, value in header_data.items()])
```

Comment:
If using `writelines` do we actually need a trailing `\n`? I haven't checked but doesn't this give us double terminators?

Reply:
Ah no I see we're passing `newline=""` to `open()` - my bad...